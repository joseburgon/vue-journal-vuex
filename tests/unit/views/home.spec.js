import { shallowMount } from '@vue/test-utils'
import Home from '@/views/HomeView'

describe('Home view tests', () => {
  test('Should render the component correctly', () => {
    const wrapper = shallowMount(Home)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Click on button should redirect to no-entry', () => {
    const mockRouter = {
      push: jest.fn()
    }

    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })

    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
  })
})