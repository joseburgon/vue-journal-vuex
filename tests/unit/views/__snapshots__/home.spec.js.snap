// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view tests Should render the component correctly 1`] = `
<main>
  <div class="px-4 py-5 my-5 text-center home"><img class="d-block mx-auto mb-4" alt="Vue logo" src="">
    <h1 class="display-5 fw-bold">Local Vue App</h1>
    <div class="col-lg-6 mx-auto">
      <p class="lead mb-4">Application to keep track of a diary, all worked with OptionsAPI and Vuex</p>
      <div class="d-grid gap-2 d-sm-flex justify-content-sm-center"><button type="button" class="btn btn-outline-primary btn-lg px-4 gap-3">Daybook Module</button></div>
    </div>
  </div>
</main>
`;
